pkgs/tools/networking/sing-box/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "sing-box";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    owner = "SagerNet";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-CNy+C5E5iAZHZ7PsS0Hj43irCuCvy/bes3kovvH81/o=";
+  };
+
+  vendorHash = "sha256-fUHfvqzbu2P7N413dDuV41myhReNSYvgF+Cc6SgG6y4=";
+
+  tags = [
+    "with_quic"
+    "with_grpc"
+    "with_wireguard"
+    "with_shadowsocksr"
+    "with_ech"
+    "with_utls"
+    "with_acme"
+    "with_clash_api"
+    "with_v2ray_api"
+    "with_gvisor"
+  ];
+
+  subPackages = [
+    "cmd/sing-box"
+  ];
+
```

Comment:
Could you add shell completions in the `postInstall` block as well, please? See https://github.com/NixOS/nixpkgs/blob/8b6d55d9695afa3a5fe2634428ad9d507ad20fa3/pkgs/tools/misc/changelogger/default.nix#L29-L34 for an example.